docs/extensions/capabilities/styling.md

```diff
@@ -0,0 +1,104 @@
+
+For layout tasks Lens is using [flex.box](https://www.npmjs.com/package/flex.box) library which provides helpful class names to specify some of the [flexbox](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Basic_Concepts_of_Flexbox) properties. For example, `div` with class names:
+```
+<div className="flex column align-center"></div>
+```
+at the end will have following css properties:
+```
+div {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+}
+```
+
+However, feel free to use any styling technique or framework like [Emotion](https://github.com/emotion-js/emotion) or just plain CSS if you prefer.
+
+## Themes
+Lens using 2 built-in themes located in `src/renderer/themes` folder each for light and dark color schemes. Active theme can be changed in the `Preferences` page.
+
+[theme selector]
+
+When Lens gets loaded it transforms selected theme `json` file into list of [CSS Custom Properties (CSS Variables)](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) which then gets injected into `:root` element so any of the down-level components can use them.
+
+[css vars listed in devtools]
+
+When user changes a theme, the process is repeated, new css variables appear instead of previous ones.
+
+If you want to follow a selected theme to keep the 'native' Lens look and feel, respecting the light/dark appearance of your extension, you can use provided variables and build-in lens components such as buttons, dropdowns, checkboxes etc.
+
+## Injected styles
+Every extention is affected by list of default global styles defined in `src/renderer/components/app.scss`. These are basic browser resets like setting `box-sizing` property for every element, default text and background colors, default font size, basic headings visualisation etc.
+
+Extension may overwrite them if needed.
+
+## Variables to use
+
+### vars.scss
+In `src/renderer/components/vars.scss` there are list of predefined variables mostly for dimensions and fonts.
+```
+// Dimensions
+$unit: 8px;
+$padding: $unit;
+$margin: $unit;
+$radius: ceil($unit * .3);
+
+// Fonts
+$font-main: 'Roboto', 'Helvetica', 'Arial', sans-serif !default;
+$font-monospace: Lucida Console, Monaco, Consolas, monospace;
+$font-size-small: floor(1.5 * $unit);
+$font-size: floor(1.75 * $unit);
+$font-size-big: floor(2 * $unit);
+$font-weight-thin: 300;
+$font-weight-normal: 400;
+$font-weight-bold: 500;
+...
+```
+
+You can use them to set consistent paddings, define font-sizes in your SCSS files e.g.
+```
+.status {
+  padding-left: $padding * 2;
+  font-size: $font-size-small;
+}
+```
+
+### theme-vars.scss
+In `src/renderer/themes/theme-vars.scss` there are list of theme-defined colors. Most of their values are different for light and dark themes. You can use them to preserve consitent view of extension with respect of selected theme.
```

Comment:
Same here: How would I reference this in an extension's build? These all reference CSS vars, so that's much better (more accessible to a wider range of extension implementations; my extension doesn't use SCSS, BTW, it uses Emotion and Lens components).

Reply:
If you want to encourage extension developers to use SCSS, then you need:
1. to provide these SCSS files via some NPM package import;
2. document what all these variables are and where they should be used;
3. make sure that all your extension examples come with SCSS processor "built-in" to the build (none of them do, right now).

Reply:
I guess there's no urge for now to encourage developers to use exactly SCSS since the only benefit currently is to use few basic variables. I've transfered them into CSS vars and removed encouragement message.